### ISSUE TITLE
OAuth2ResourceServer configuration tests use deprecated extractAuthorities

### ISSUE THREAD

**Issue by jzheaux** [spring-projects/spring-security#6516]
`JwtAuthenticationConverter#extractAuthorities` was deprecated in #6273 via 0f7dff3.

`OAuth2ResourceServerConfigurerTests` and `OAuth2ResourceServerSpecTests` should be updated to no longer use it, changing instead to use `setJwtGrantedAuthoritiesConverter`.

**Comment by vishalvrv9** [comment]
Hey @jzheaux, haven't completely analyzed the issue but would like to work on it. Can I take this up?

**Comment by jzheaux** [comment]
Yep, it's yours!

**Comment by jzheaux** [comment]
@vishalvrv9 wanted to check in to see how things are going. Are there any questions I can answer for you?

**Comment by vishalvrv9** [comment]
Hey @jzheaux ,
firstly, apologies for the delay

As per analysis, 
- In `OAuth2ResourceServerSpecTests`, since the method signature is different from `extractAuthorities`

what should the implementation changes look like? Can you provide me a starting point as after trying to implement the `setJwtGrantedAuthoritiesConverter` I am a little lost

**Comment by jzheaux** [comment]
@vishalvrv9 no problem, my apologies for lagging in my response :)

Something like this might work well:

```diff
- JwtAuthenticationConverter converter = new JwtAuthenticationConverter() {
-				@Override
-				protected Collection<GrantedAuthority> extractAuthorities(Jwt jwt) {
-					String[] claims = ((String) jwt.getClaims().get("scope")).split(" ");
-					return Stream.of(claims).map(SimpleGrantedAuthority::new).collect(Collectors.toList());
-				}
-			};
+ JwtAuthenticationConverter converter = new JwtAuthenticationConverter();
+ converter.setJwtGrantedAuthoritiesConverter(jwt -> {
+ 		String[] claims = ((String) jwt.getClaims().get("scope")).split(" ");
+		return Stream.of(claims).map(SimpleGrantedAuthority::new).collect(Collectors.toList());
+ });
```

**Comment by jzheaux** [comment]
@vishalvrv9 Is this something that you'd still like to work on?

**Comment by vishalvrv9** [comment]
Hey @jzheaux, Yeah definitely. Apologies lost track of it. Will try submitting a PR as soon as possible.

**Comment by vishalvrv9** [comment]
@jzheaux have submitted PR #6904

**Comment by jzheaux** [comment]
Thanks, @vishalvrv9! I left a bit of feedback in the PR.

**Comment by vishalvrv9** [comment]
Hey @jzheaux , worked on the feedback a couple of days back.

**Comment by jzheaux** [comment]
Fixed via b6e8997
